sklearn/tests/test_naive_bayes.py

```diff
@@ -859,6 +864,43 @@ def test_alpha():
         m_nb.partial_fit(X, y, classes=[0, 1])
 
 
+def test_check_alpha():
+    """The provided value for alpha must only be
+    used if alpha < _ALPHA_MIN and force_alpha is True.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/10772
+    """
+    _ALPHA_MIN = 1e-10
+    b = BernoulliNB(alpha=0, force_alpha=True)
+    assert b._check_alpha() == 0
```

Comment:
May we go through a public API, such as `fit` for testing alpha?

Reply:
Actually I don't think that `alpha`, _after_ `_check_alpha` is applied is ever publicly exposed? Unless I'm missing something

Reply:
I created a private `_alpha` attribute to account for this (imo it's a bit more consistent w/ other estimators now too).